pymodbus/transaction.py

```diff
@@ -368,16 +374,13 @@ def _transact(self, request: ModbusRequest, response_length, full=False, broadca
             packet = self.client.framer.buildPacket(request)
             Log.debug("SEND: {}", packet, ":hex")
             size = self._send(packet)
-            if (
-                isinstance(size, bytes)
-                and self.client.state == ModbusTransactionState.RETRYING
-            ):
+            if self.client.state == ModbusTransactionState.RETRYING:
```

Comment:
This change do not look correct, we only want to enter the "if" if we received something.

This have nothing to do with typing, and it is not covered by tests.

Reply:
I removed the `isinstance(size, bytes)` condition, which based on the return type of `self._send(packet)` is always an integer. This was flagged by the type checker.

Reply:
But it still needs to get argumented and tested. I see quite a different behavior, because you return a fixed b"", which basically never should be returned in a retry situation.

Reply:
> I see quite a different behavior, because you return a fixed b"", which basically never should be returned in a retry situation.

I removed the return entirely and changed this check to verify returned size is not 0.